### ISSUE TITLE
How about the result of SSD with focal-loss?

### ISSUE THREAD

**Issue by bailvwangzi** [zimenglan-sysu-512/Focal-Loss#4]
I try to merge your focal-loss with SSD, the training loss can converge to 1.x, but the testset mAP is not as well as OHEM method. Have you test focal-loss on any dataset?

**Comment by zimenglan-sysu-512** [comment]
Hi @bailvwangzi,

I only test focal-loss on my own dataset, found that it drops a little performance. so sad.

**Comment by birdwcp** [comment]
@bailvwangzi 
  Did you use  MinehardExamples()  when merged focal-loss with SSD ?

**Comment by bailvwangzi** [comment]
@birdwcp 
I just change mining_type to NONE. So MinehardExamples() will not do hard example mining, just compute num_matches.

**Comment by zimenglan-sysu-512** [comment]
hi @bailvwangzi,
`alpha = 1` and `gamma=3` can boost 1.4% mAP on pascal voc 2007 testset using `pvanet`.

**Comment by bailvwangzi** [comment]
@zimenglan-sysu-512  i have some questions,
1.how many iterations have you trained? In the first few iterations the loss can be very large or NAN?
2.how to initialize the network? just use imagenet to finetune?
3.the final loss = ?
thanks!

**Comment by bailvwangzi** [comment]
@zimenglan-sysu-512 you use the two-stage method pvanet? not ssd?

**Comment by zimenglan-sysu-512** [comment]
hi @bailvwangzi,
it's my friend's results. yes, use two stages (PVANET), in which we replace the `softmax loss` with `focal loss` in `Fast RCNN` instead of `RPN`.

**Comment by zimenglan-sysu-512** [comment]
@bailvwangzi, since it's my friend's results, so i can't give you the loss, and the hyper-parameters.

**Comment by bailvwangzi** [comment]
@zimenglan-sysu-512  since pvanet not use OHEM, i think the 1.4% mAP boost is unfair to say focal loss is better than OHEM.

**Comment by zimenglan-sysu-512** [comment]
hi @bailvwangzi,
here i don't compare it with ohem, i just want to prove `focal loss` works or not.

**Comment by bailvwangzi** [comment]
I think focal loss can be better than _softmaxWithLoss_ or _sigmoidCrossEntropyLoss_ , but it's more meaningful to prove focal loss is better than ohem, since the paper get significant improvements. 
![image](https://user-images.githubusercontent.com/18113049/30318495-d8942134-97df-11e7-9058-4cf91ac9fe43.png)

**Comment by cyliu7** [comment]
hi, @zimenglan-sysu-512 .
In two stage network, there are two softmaxloss during training, RPN's softmaxloss and Fast rcnn's softmaxloss. which loss did you replace with focal loss?

**Comment by zimenglan-sysu-512** [comment]
hi @liuchy666,
i replace the softmax loss of RCNN with focal loss.
